### PR TITLE
go.*: Remove the stale github.com/Sirupsen/logrus replace directive

### PR DESCRIPTION
**Pull request by timflannagan**
# Description

A quick maintenance PR for removing a stale replace directive. This was previously needed _way_ back in time when the original logrus repo was renamed. See https://github.com/sirupsen/logrus/issues/570 for more details.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Implausiblyfun** [comment]
can you please attach the go mod graph before and after to show that this replace is not in affect?

**Comment by timflannagan** [comment]
@Implausiblyfun Just to quickly clarify: for the mod graph, what do you suggest? Running `go mod why ...` locally and pasting the output or did you have something else in mind? In either case, I'm happy to do that tomorrow and paste the output as a comment when I get back to my comp.

With that said, I'm now wondering whether the current evidence, which is running `go mod tidy` doesn't result in any go.sum changes, and `cat`ing the go.sum file itself  locally doesn't contain this directive could be concluded as sufficient evidence? AFAIK, this used to be a problem when the repository was renamed, but indirect repositories that consume this library hadn't updated their go.mod files, and it was causing wide spread conflicts. Since then, the ecosystem has adjusted as we can remove the replace pin to get some short term maintenance wins.

**Comment by nfuden** [comment]
yeah just something quick to show that the replace has no effect therefore the removal is a net win.

File: go.mod [hunk 1]
```diff
@@ -326,7 +326,6 @@ require (
 )
 
 replace (
-	github.com/Sirupsen/logrus => github.com/sirupsen/logrus v1.4.2
```
Comment (sam-heilbron) [inline go.mod:329]: If we are removing this here, can we also be sure to remove it https://github.com/solo-io/solo-projects/blob/main/go.mod#L503?

**Review by sam-heilbron — APPROVED** [review]

**Review by davidjumani — APPROVED** [review]

**Comment by timflannagan** [comment]
@nfuden I ran this from a different branch, but here's the why output:

```bash
$ dev :: /work/gloo ‹chore/update-contributing› » go mod why github.com/Sirupsen/logrus      
# github.com/Sirupsen/logrus
(main module does not need package github.com/Sirupsen/logrus)
```

I think this is sufficient along with `go mod tidy` producing no go.sum changes, and the go.sum file doesn't contain this entry either.

**Pull request merged**